androlo/bytes/ByteSlice.sol

```diff
@@ -0,0 +1,171 @@
+/// Used to create slices from 'bytes'.
+contract ByteSlice {
+
+    struct Slice {
+        uint _unsafe_memPtr;   // Memory address of the first byte.
+        uint _unsafe_len;      // Length.
+    }
+
+    /// Create a slice from bytes.
+    function fromBytes(bytes memory bts) internal constant returns (Slice memory slice) {
+        uint memPtr;
+        assembly {
+            memPtr := add(bts, 0x20)
+        }
+        slice = Slice(memPtr, bts.length);
+    }
+
+    /// Length of the slice.
+    function len(Slice memory slice) internal constant returns (uint) {
+        return slice._unsafe_len;
+    }
+
+    /// returns the byte at 'index'.
```

Comment:
I will add more documentation and tag returns and params etc., though it would be great if 
@returns could handle multiple return values, like params does, so it can be used for all possible types of return values.